jbmc/src/java_bytecode/java_types.cpp

```diff
@@ -333,14 +333,40 @@ std::vector<typet> parse_list_types(
   const std::string class_name_prefix,
   const char opening_bracket,
   const char closing_bracket)
+{
+  // Loop over the types in the given string, parsing each one in turn
+  // and adding to the type_list
+  std::vector<typet> type_list;
```

Comment:
:pick: 
```C++
const std::vector<std::string> type_strings = parse_raw_list_types(src, opening_bracket, closing_bracket); 
std::vector<typet> type_list;
std::transform(
  type_strings.begin(), 
  type_strings.end(), 
  std::back_inserter(type_list), 
  [&](const std::string &raw_type) {
    const typet new_type = java_type_from_string(raw_type, class_name_prefix);
    INVARIANT(new_type != nil_typet(), "Failed to parse type");
    return new_type;
});
return type_list;
```

Reply:
Does this actually give any real benefit when you're simply looping through an entire list with no bells or whistles?

Reply:
Cheers, was mostly just curious about the recommendation (wasn't meant as a refutal).